crates/sui-framework/packages/dwallet-system/sources/sui_state_proof.move

```diff
@@ -82,15 +90,18 @@ module dwallet_system::sui_state_proof {
         new_checkpoint_summary: vector<u8>,
         ctx: &mut TxContext,
     ) {
-        let committee_verified_bytes = sui_state_proof_verify_committee(prev_committee.committee, new_checkpoint_summary);
+        let (new_committee_verified_bytes, committee_epoch) = sui_state_proof_verify_committee(prev_committee.committee, new_checkpoint_summary);
 
         let committee_new = EpochCommittee {
                                     id: object::new(ctx),
-                                    committee: committee_verified_bytes,
+                                    committee: new_committee_verified_bytes,
                                     };
 
         registry.highest_epoch = registry.highest_epoch + 1;
 
+        assert!(committee_epoch == registry.highest_epoch, EWrongEpochSubmitted);
```

Comment:
This should be a good fix, however, it is best practice to perform security checks _before_ updating state etc. 

Although it should be ok as a failed assert will revert state changes, I suggest moving this check right after verifying the state proof, and before creating a new committee object and mutating the registry.